src/rubyfmt.rb

```diff
@@ -716,6 +716,12 @@ def format_int(ps, rest)
   ps.emit_indent if ps.start_of_line.last
 
   int = rest[0]
+  int_chars = int.chars
+
+  if int_chars[0] != "0" && !int_chars.detect { |c| c == "_" }
```

Comment:
If we want to be fancy, we could use `none?` rather than `![].detect`.

Reply:
We should probably go with `!int_chars.include?("_")` — it's less code than `#detect` or `#none`, and something like an order of magnitude faster.